### DISCUSSION TITLE
Enable Link-Time Optimization (LTO)

### DISCUSSION THREAD

**Discussion by zamazan4ik** [ki-editor/ki-editor#321] [Ideas]
Hi!

I noticed that in the Cargo.toml file Link-Time Optimization (LTO) for the project is not enabled. I suggest switching it on since it will reduce the binary size (always a good thing to have) and will likely improve the application's performance a bit (in this case, it's not critical at all but anyway).

I suggest enabling LTO only for the Release builds so as not to sacrifice the developers' experience while working on the project since LTO consumes an additional amount of time to finish the compilation routine. If you think that a regular Release build should not be affected by such a change as well, then I suggest adding an additional release-lto profile where additionally to regular release optimizations LTO also will be added. Such a change simplifies life for maintainers and others interested in the project persons who want to build the most performant version of the application. Using ThinLTO also should help).

Thank you.

**Comment by wongjiahau** [comment]
Hey @zamazan4ik, thanks for the suggestion! 

Unfortunately, I’m not well-versed at all in this topic, would you like to submit a PR for it?

**Reply by zamazan4ik** [reply]
Sure - here we go: https://github.com/ki-editor/ki-editor/pull/324

**Comment by setop** [comment]
Does it have any positive or negative impact one the build time ?
Currently on my machine, full built takes 35 minutes.

**Reply by wongjiahau** [reply]
I’m not sure about this as I haven’t tried it before
